### ISSUE TITLE
Attach logo in emails only if needed by template

### ISSUE THREAD

**Issue by jaime-0000** [PrestaShop/PrestaShop#27645]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

Related to https://github.com/PrestaShop/PrestaShop/issues/19070
It is frustrating that the Mail class embeds the logo into every email if either the mail logo or the site logo is defined.

Some emails and email templates do not need this embedded logo which is a waste of bandwidth.  

Some email clients such as gmail actually display it as an attachment instead of hiding it as an embedded file, which is undesirable.

Some of the emails generated by ps_emailalerts are not for customers, and would benefit from not having this logo embedded

https://github.com/PrestaShop/PrestaShop/blob/7aabf1874574aa565e6a3e8dd619701178cbb8c5/classes/Mail.php#L527
```php
            if (false !== Configuration::get('PS_LOGO_MAIL') &&
                file_exists(_PS_IMG_DIR_ . Configuration::get('PS_LOGO_MAIL', null, null, $idShop))
            ) {
                $logo = _PS_IMG_DIR_ . Configuration::get('PS_LOGO_MAIL', null, null, $idShop);
            } else {
                if (file_exists(_PS_IMG_DIR_ . Configuration::get('PS_LOGO', null, null, $idShop))) {
                    $logo = _PS_IMG_DIR_ . Configuration::get('PS_LOGO', null, null, $idShop);
                } else {
                    $templateVars['{shop_logo}'] = '';
                }
            }
            ShopUrl::cacheMainDomainForShop((int) $idShop);
            /* don't attach the logo as */
            if (isset($logo)) {
                $templateVars['{shop_logo}'] = $message->embed(\Swift_Image::fromPath($logo));
            }
```

### Describe the solution you'd like

The Mail class could include an additional optional parameter such as

```php
@param bool $embedLogo file
```

with a default value of true for backwards compatibility.

Then just embed the logo if the parameter is true:
https://github.com/PrestaShop/PrestaShop/blob/7aabf1874574aa565e6a3e8dd619701178cbb8c5/classes/Mail.php#L526
```php
            if (isset($logo) && $embedLogo) {
                $templateVars['{shop_logo}'] = $message->embed(\Swift_Image::fromPath($logo));
            } else {
                $templateVars['{shop_logo}'] = '';
            }
```

### Alternatives you've considered

Overriding Mail class, which is not ideal.

### Do you plan to work on this feature?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by khouloudbelguith** [comment]
Hi @jaime-0000,

Thank you for your suggestion.
Isn't your issue similar to this ticket: https://github.com/PrestaShop/PrestaShop/issues/19070?.

Thank you!

**Comment by jaime-0000** [comment]
Hi, @khouloudbelguith thanks for picking this up so quickly. 

It is indeed related, but I believe it not the same request. I agree with the other request in that plain text emails should not embed the logo image, but this feature request is much more generic. There are use cases where you really do not need the embedded logo, even in html emails, but it should be up to each implementation of the Mail class to decide. 

Some email templates may require the logo and others may not. This feature request just aims to add a little extra flexibility for module and template developers, with a very small change impact.  A global setting (embed logo in emails or not) is not something I would consider useful, as even in one single shop instance only certain emails may require the logo (ie. those directed to customers), and others should be able to opt out (ie. internal emails, low stock alerts, new orders to be processed, etc.).

**Comment by jaime-0000** [comment]
Also, implementing a method in the class to embed additional files, returning the cid, would also be very useful, as some mail templates would benefit from embedding additional images instead of linking them (such as product images). But that should probably be another feature request.

**Comment by khouloudbelguith** [comment]
@jaime-0000, thanks a lot for your feedback.
The Product Team will take it into consideration for future developments. 

Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you or anyone else is willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know!

Thank you!

**Comment by jaime-0000** [comment]
I'd be happy to try to contribute a PR.

**Comment by cruftex** [comment]
Specs:
- Existing templates using `shop_logo` variable work as before
- For templates not using the `shop_logo` variable, no logo is attached
- No additional configuration parameter is added

Idea:

- `$magicForLogo = 'AzcvdSD89734'`
-  `$templateVars['{shop_logo}'] = $magicFogLogo`
- Render mail
- Check whether mail contains the string `$magicForLogo`
- if yes, attach logo and replace string `$magicForLogo` with logo attachment name

I also suggest some polishing around the logo attachment code in `Mail.php`.

**Comment by cruftex** [comment]
Please change title of this issue, currently it is:

> Mail class should include parameter that allows opting out of embedding logo file

To

> Attach logo in emails only if needed by template

**Comment by krodelabestiole** [comment]
why not at least an option to link the logo from the website instead of filling up millions of email boxes with redundant attachments ?

**Comment by cruftex** [comment]
@krodelabestiole 

> why not at least an option to link the logo from the website instead of filling up millions of email boxes with redundant attachments?

You can just put links to images in the template. You do not need that as an "option". You have it already.
Only thing to do is to remove the attached image, if its not used. That's why this issue.

**Comment by Hlavtox** [comment]
@cruftex This would be good to do. I sometimes send empty html template just with some custom messages, and the logo is always attached for no reason. :-)

**Comment by cruftex** [comment]
@Hlavtox 
Thanks for the title change.

Also I wrote Specs and this as still the label "Needs Specs". No idea how the process of solving such a tiny problem will be.

Not so keen to do a PR, since other PRs are stuck and I don't see a sense to add more that do not lead somewhere.

**Comment by Hlavtox** [comment]
@cruftex What can I say, I spent 4 hours on Saturday closing like 300 old issues, fixing labels, responding etc. 

How hard can it be to make a monitoring for the issues, for example - if an issue has label X but not X, warn about it.

Regarding the PRs, yes, but we now have more committers that actually have energy to review other PRs other than their colleagues, like @Codencode :-) I am sure that if you make a PR where my technical skills will be enough to understand it, it won't get stuck.

Cc @ibahloul-ps @cnavarro-prestashop Guys, if you want to encoursge contributions, I suggest not adding more bureaucracy, but making sure that somebody is taking care of the project, other than @kpodemski. It's a shame that skilled people like @cruftex get pissed off from contributing.

**Comment by Codencode** [comment]
@Hlavtox, thanks for the shout-out, it's nice to see that someone notices what you do :-)
Of course, the admiration is mutual!!!

**Comment by kpodemski** [comment]
@Hlavtox @cruftex @Codencode 

You may have already seen our recent “Call for Contributions.” It’s part of how PrestaShop SA supports and amplifies community involvement in the project. We focus on one area instead of "being everywhere".

Since PrestaShop is an open project, we rely heavily on committers and contributors who can review and merge pull requests to keep things moving. This is one of the most impactful ways the community can help the project scale, especially as the number of PRs continues to grow.

And speaking more personally, @Hlavtox: I know you care deeply about your contributions and naturally want to see them reviewed quickly. That’s completely normal - most open-source contributions start from a very personal need, and there’s nothing wrong with that. At the same time, the long-term sustainability of the project depends on **having more people reviewing and unblocking each other’s work**, not just submitting new PRs.

What I can do on my side, and I'm trying my best, is work on unblocking some of the topics that are currently stuck and encourage more contributors to step into reviewing. The more committers actively participating in this part of the process, the healthier and faster the project becomes for everyone. Thank you for the kind words, but you have to know that I'm doing it mostly in my free time, so you cannot compare me to other PrestaShop SA employees. I'm working with PrestaShop SA as a contractor, but _gardening_ the project is not what PrestaShop SA is paying me for (at least not in the parts of the project you'd like me to see ;-))

Regarding PrestaShop SA’s involvement: the internal team is concentrated on several major initiatives, the new discounts page, the new API, One Page Checkout, Hummingbird 2, the ongoing price-calculation rework happening behind the scenes (and more). On top of that, there’s regular maintenance, releases, and the upkeep of the tools the project relies on... 😱  It’s a substantial workload, and we simply cannot be everywhere at once. We’re doing our best with the resources we have.

@Hlavtox You may not always agree with the company’s priorities, yes, whether it’s focusing on discounts or another big project, and that’s perfectly fine. Just as you choose to invest your time based on your clients’ needs, the company invests its resources where it believes they will deliver the most impact. The open-source model gives everyone room to contribute where they feel it matters most. 

So rather than focusing on what PrestaShop SA “should” do, it’s more productive to share feedback constructively. We do listen, and it’s valuable. But it becomes counter-effective when there’s an expectation for the company to invest more in the project while simultaneously criticizing every effort made to strengthen the business and generate the revenue that ultimately funds that investment. Those two positions can’t really coexist, and finding a balanced perspective helps the whole ecosystem. I hope that one day you'll understand that.

**Comment by cruftex** [comment]
@kpodemski 

Thanks for you very detailed and also personal comment.

I totally agree that there will be different interests and driving forces. Also it is important to focus developers at some areas. Task switching kills productivity.

Out of frustration I decided to keep communication and involvement with the PrestaShop project rather limited.

I am a bit surprised to see the long comment from you. This is very valuable. I took the time the whole morning to write this comment, although I have a lot other things to do urgently. 

Hopefully, it finds you well and is helpful.

> That’s completely normal - most open-source contributions start from a very personal need

I fixed various problems in PS, to make it work for us. Doing the Open Source contribution is extra work put in to make it work for others as well. This is not my personal need but my ethics. I cannot expect only to receive.

> we rely heavily on committers and contributors

If you (you, meaning PrestaShop SA) does, then building a healthy community, should be a priority as well. That should not be just your personal good will in the free time.

> The more committers actively participating in this part of the process, the healthier 

Which process?

> ongoing price-calculation rework happening behind the scenes

Great point. Regarding price calculations I made several attempts to contribute:

- https://github.com/PrestaShop/PrestaShop/discussions/32969
- https://github.com/PrestaShop/PrestaShop/discussions/36324

I met personally for a day with a PrestaShop SA employee to go through it and explained basic accounting and tax calculation principles. The task for the employee was to create a "specification" what needs to be done. The specification was written. Then the employee got sacked. However, nobody told me that. My feedback mails, although some other PrestaShop SA employees were on CC, just went unanswered.

In 2024 @matks, made a comment (https://github.com/PrestaShop/PrestaShop/discussions/32969#discussioncomment-9007624), saying that it is complex and PS cannot fix everything.

Seeing that PS does not have the capacity or willingness to address a core problem, I decided to try once again and at least do a minimal step to improve:

- https://github.com/PrestaShop/PrestaShop/pull/38805

It is always better to discuss over code, rather over lengthy specs and requirements.

You were commenting in this PR. You are assigned as a reviewer and GitHub tells me that you are the one that needs to review and tick it of. You did not answer to my comment from May 3. Exactly half a year now.

To be clear, although you (@kpodemski) are personally involved, I don't mean it as a personal critique. I would probably have done the same. We all have a lot on the plate and if nobody else takes interest, maybe it might not be worth attention.

Usually, I am more persistent and would write a "ping!" after two weeks or so, however, ultimately, I was sad that nobody else seemed to take interest in it.

@ibahloul-ps (the PrestaShop open-source project's product manager) was mentioned, but did not comment.

**Now, PrestaShop SA is working on that "behind the scenes"?!**

I try to be calm, rational and constructive. However, out of frustration, I cannot help to put in some irony in the next paragraph:

_Why this happened as it happened, I can only speculate. One speculation might be that PrestaShop SA just wants to steer major topics by themselves and does not put their resources on something coming from the community. Even if it is a major problem, even if it makes sense. Maybe it is too embarrassing to give somebody else credit to help fix the price calculation bugs which should an eCommerce company be capable of on its own? Is it ego? Well, how embarrassing is it on the other hand to get repeated bug reports each month for invoices being off by a cent? How embarrassing is it to have a "Rounding Mode" selector in an eCommerce application that has no tests, no exact defined semantics and no one can say what its good for? In a "leading eCommerce" system? For decades? Really? Well, you cannot be everywhere. And another speculation: Maybe there are some PrestaShop developers tasked to solve a problem, but not to interact with the public community about it?_

Back to the constructive part:

> the internal team is concentrated on several major initiatives, the new discounts page, the new API, One Page Checkout, Hummingbird 2, the ongoing price-calculation rework happening behind the scenes

Whenever some contributor is opening a bug or does a suggestion, you mandate for specs. Where are the specs on these initiatives and the community feedback on that?

There are several systemic and cultural problems at play here.

Being afraid to touch code because nobody knows what it does, or asking for detailed upfront specifications are one of the many anti patterns I could talk of.

Asking for specification is actually worse in two ways: It's a waste and not inclusive. And it is "not inclusive" in many ways  when you are constantly changing or not communicating what a spec is and don't lead by example yourself.

But, let's stay with the Elephant(s) in the room:

I was attending the first developer meeting. The concluding panel discussion started with a lively chat in French for 15 minutes. Of course, those who did not speak French, and there were many, had no idea what was talked about. After I spoke up the organisers apologised. Still, this not inclusive and disrespectful to those travelling the longer way to Paris and also don't speak English as their first language. I assume that was improved on the next conference, however, it serves well as an example.

Within the PrestaShop project, unfortunately, there are several layers of communication circles and information silos. Like the slack server. The "inner circle" is close nit and enjoys speaking in French, I assume. If one company's project is mainly based on building an international Open Source project and community, the language for all project discussions has to be English.

From what I see, PrestaShop SA is working the same like companies not doing Open Source projects and operating in a traditional waterfall manner. Management is talking to employees what features they should build (in French). When ready, these features are thrown over the fence to the public community, which also includes their customers. 

**Having an open community and building secretly does not go together**.

For an Open Source community you have to **build in the open** and every person involved needs to **know and follow the same rules** and be **willing, allowed and encouraged to communicate publicly**.

**Comment by Codencode** [comment]
@kpodemski
Krystian, I totally agree with you. Of course we would all like PRs to be reviewed as quickly as possible, but we fully understand that this isn’t always feasible, especially considering all the other priorities involved.

I’d like to clarify one thing: at the moment I haven’t yet built a full e-commerce project with version 9, but I will soon. Most of the PRs I’ve opened so far (on v9) come from issues reported by other users or from problems I happened to notice while testing the new version.

I also want to emphasize that this is not a criticism of what you said — on the contrary, I completely agree with your point. It’s simply a personal reflection I wanted to share 😃.

**Comment by kpodemski** [comment]
@cruftex 

> If you (you, meaning PrestaShop SA) does, then building a healthy community, should be a priority as well. That should not be just your personal good will in the free time.

The company did a lot to make the project more transparent and open, and I'm not sure if more reviews coming from PrestaShop SA developers are a must-have to build a healthier community. I firmly believe that it's about giving tools to empower the community.

> You were commenting in this PR. You are assigned as a reviewer and GitHub tells me that you are the one that needs to review and tick it of. You did not answer to my comment from May 3. Exactly half a year now.

It's because I simply don't like the proposed solution. Hiding rounding options doesn't solve the underlying issue, also, some of the modules recommend switching this configuration to something else, there are also certain stores with certain settings that require using a different option.

> Now, PrestaShop SA is working on that "behind the scenes"?!

Let me be more precise. Your work, work from Rosa, and other discussions regarding price calculation **are part of this process**. The process of ensuring that everything is taken into account, and the project can start with specific tasks for developers to follow. Your help has been invaluable. We will share more details when the time comes.

> Why this happened as it happened, I can only speculate. One speculation might be that PrestaShop SA just wants to steer major topics by themselves and does not put their resources on something coming from the community. Even if it is a major problem, even if it makes sense. Maybe it is too embarrassing to give somebody else credit to help fix the price calculation bugs which should an eCommerce company be capable of on its own? Is it ego? Well, how embarrassing is it on the other hand to get repeated bug reports each month for invoices being off by a cent? How embarrassing is it to have a "Rounding Mode" selector in an eCommerce application that has no tests, no exact defined semantics and no one can say what its good for? In a "leading eCommerce" system? For decades? Really? Well, you cannot be everywhere. And another speculation: Maybe there are some PrestaShop developers tasked to solve a problem, but not to interact with the public community about it?

No, no. Please don't understand it like this. It's because certain preparations need to be made internally so the project can later be put in place publicly. If Illies works on the project, he doesn't document every single step publicly. He doesn't have to for the project to be transparent. Specs will be shared, exactly as they were regarding Discounts.

You should also know that if you follow the project closely, we try to give more visibility to the community, agencies, and individual contributors. We have interviewed the contributors numerous times, created a Wall of Fame in the back office,  mentioned top contributors during Live Updates, and some were mentioned during the 2025 talk at the Developer Conference. I'm very happy when I can say that some projects have been implemented with the help of the community, and this has happened numerous times, including @Hlavtox and @Codencode.

**Let me reassure you, nothing is kept "in secret" just to get the credit for the work. That's absolutely not the case.**

> Whenever some contributor is opening a bug or does a suggestion, you mandate for specs. Where are the specs on these initiatives and the community feedback on that?

That is not true. They are not **required** for the pull request to be approved. A few examples from PrestaShop SA side:
https://github.com/PrestaShop/PrestaShop/issues/38644
https://github.com/PrestaShop/PrestaShop/issues/40218

Of course, some work **is** internal. We could discuss for hours about the details, why, how, when, what for, etc. The most important part of the process is submitting the pull request, and that is always public.

> I was attending the first developer meeting. The concluding panel discussion started with a lively chat in French for 15 minutes. Of course, those who did not speak English, and there were many, had no idea what was talked about. After I spoke up the organisers apologised. Still, this not inclusive and disrespectful to those travelling the longer way to Paris and also don't speak English as their first language. I assume that was improved on the next conference, however, it serves well as an example.
> Within the PrestaShop project, unfortunately, there are several layers of communication circles and information silos. Like the slack server. The "inner circle" is close nit and enjoys speaking in French, I assume. If one company's project is mainly based on building an international Open Source project and community, the language for all project discussions has to be English.

I was there, I remember that, and I'm sorry you feel that way. What can I say? I will be completely honest with you: I hope that one day we'll have a true international developer conference, because I also don't speak French, and that conflicts with what we're trying to create in this project: encouraging more developers from around the world to work with PrestaShop.

I've been in this ecosystem for many years. I've seen the project from the outside and from the inside, and it gives me a unique perspective. It helps me understand certain things better.

Let me emphasize again: the PrestaShop team genuinely strives to do their best. We have talented individuals at PrestaShop, and over the past few years, we've completed some amazing projects. I understand there's often a sense of it being too little or too slow, with some wanting more visibility or different features on the roadmap, or more reviews, etc. We cannot have everything at once, even tho the company is **by far** the biggest sponsor of the project.

**Comment by Hlavtox** [comment]
@kpodemski @cruftex 

> The company did a lot to make the project more transparent and open...

It did, but then took it back gradually with many actions.

> No, no. Please don't understand it like this. It's because certain preparations need to be made internally so the project can later be put in place publicly. If Illies works on the project, he doesn't document every single step publicly. He doesn't have to for the project to be transparent. Specs will be shared, exactly as they were regarding Discounts.
> 
> You should also know that if you follow the project closely, we try to give more visibility to the community, agencies, and individual contributors. We have interviewed the contributors numerous times, created a Wall of Fame in the back office, mentioned top contributors during Live Updates, and some were mentioned during the 2025 talk at the Developer Conference. I'm very happy when I can say that some projects have been implemented with the help of the community, and this has happened numerous times, including @Hlavtox and @Codencode.
> 
> Let me reassure you, nothing is kept "in secret" just to get the credit for the work. That's absolutely not the case.
> 
> Of course, some work is internal. We could discuss for hours about the details, why, how, when, what for, etc. The most important part of the process is submitting the pull request, and that is always public.

The problem is that many of the initiatives coming from project management or the development team feel disconnected from real-world needs. And not in small things. Early discussion with merchants and long-time contributors could prevent unnecessary work and misunderstandings. Since Mateus left, nobody wants to discuss anything - I understand, we are only running ecommerce store for 20 years - not relevant. :-) Or, nobody wants to discuss anything with @ShaiMagal, one of the most experienced people with PrestaShop that there are. If the preferred approach is to let issues surface only once a PR is submitted and then blocked, okay.

Also, having a PM that doesn't reply to absolutely anything is completely useless - no answers on GitHub, no replies on Slack, no effort to consult with the community or gather real feedback. This does not build trust, does not strengthen collaboration, and certainly does not help the project move in a direction aligned with real-world needs.

> I was there, I remember that, and I'm sorry you feel that way. What can I say? I will be completely honest with you: I hope that one day we'll have a true international developer conference, because I also don't speak French, and that conflicts with what we're trying to create in this project: encouraging more developers from around the world to work with PrestaShop.

@cruftex This part is actually very simple - it’s a French company, discussing things with French agencies. Others are expected to adapt to that. Specifications and issues of some people written in French only reinforce the impression that, from their perspective, the community doesn’t really exist or doesn’t matter.

**Comment by BenTen** [comment]
Once again, a conflict in the PrestaShop ecosystem…

I believe everyone has now had the opportunity to say what was on their mind.
Before this goes any further, maybe we can all take a short break and reflect.

Just a friendly reminder: this issue is titled "Attach logo in emails only if needed by template", not "Let’s have a 300-comment debate on national stereotypes and whether French people are selfish in groups" 😅

Perhaps we could get back to the actual topic (or move the sociology discussion elsewhere) and find a solution that actually helps the project and the community.

Sorry.

**Comment by cruftex** [comment]
@BenTen 
>  "Let’s have a 300-comment debate on national stereotypes and whether French people are selfish in groups"

That is not what I meant and said. It's not helpful to spin it that way and it hurts my feelings.

Everybody is more comfortable speaking their mother tongue.

Also, if somebody does not understand English very well, those have to be integrated in the discussion as well. It is still in Paris.

It is about creating an inclusive environment for everyone, as the Code of Conduct for the community mandates.

> Perhaps we could get back to the actual topic

If somebody has something to add, I will read it. However, yes, its not a good time and place for broad discussions.

@kpodemski 
>  It's because certain preparations need to be made internally

I am long enough in business to know the many reasons why that is. However, if there is a will, it can be addressed and overcome.

**Comment by cruftex** [comment]
@kpodemski 
Many thanks for your time commenting here. Its good to hear your perspective and its good for my mental health :)

**Comment by kpodemski** [comment]
@cruftex Jens, I really appreciate your feedback and observations. If you want to discuss anything, I'm available. I truly believe that, for the project's well-being, we should be open to discussing with the community. I'm always happy to do that.

**Comment by boo-code** [comment]
Still the case on 9.2.x. `Mail::Send()` decides from configuration rather than from the template:

```php
if (isset($logo) && $configuration['PS_MAIL_TYPE'] != Mail::TYPE_TEXT) {
    $templateVars['{shop_logo}'] = 'cid:shop_logo';
    $email->embedFromPath($logo, 'shop_logo');
}
```

so a template that never displays the logo still carries the image, and mail clients show it as an attachment on a message with no visible logo. That is the report.

Measured on the shipped templates, which is worth knowing before deciding how to fix it: all 34 html templates in `mails/en` reference `{shop_logo}`, through the shared `header.html.twig` component, and none of the 34 text templates do. So the shipped set always wants the logo, and it is custom mail themes and module templates that carry a file they never show.

That points at the condition rather than at a new setting: the template already states whether it wants the logo, so a setting would be a second place to say the same thing and a chance for the two to disagree. Checking the html body for `{shop_logo}`, or for `cid:shop_logo` if a template writes the `img` tag itself, is enough.

Two details that decide whether such a change is safe:

- The check has to sit after `actionEmailAddAfterContent`, or a module that adds the placeholder through that hook loses its logo.
- When the logo is not attached, `{shop_logo}` should resolve to an empty string rather than keep `cid:shop_logo`, otherwise a text part that happens to carry the placeholder ends up pointing at an attachment that was never added.

On the second half of the request, a way to embed arbitrary files and get the content id back: that one is a new public API on `Mail` rather than a condition on an existing one, so it is worth keeping separate from this.

The related #19070 about plain text mails is already covered by the `PS_MAIL_TYPE` check in the same condition.
